Add cidr range firewall and removed network tags

praveenramuu opened the pull request:
Added cidr range firewall and removed network tags

praveenramuu committed:
Added cidr range firewall and removed network tags

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

sasikeda commented:
- Looks, egress rules are only created. Not sure if I am missing something, could not see ingress.
- compute egress rule is also created, in storage-only mode.
- 60sec wait may not be required in gcp, need to investigate.
- In the inventory, the "storage_cluster_instance_ids": ["6394890111382239406","1037653424555473070","6095985740772587694","674473927763343534"] need to be replaced with self-link.

praveenramuu committed:
added var to bypass waiting ,removed compute egress, updated selflinks

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu commented:
> * Looks, egress rules are only created. Not sure if I am missing something, could not see ingress.
> * compute egress rule is also created, in storage-only mode.
> * 60sec wait may not be required in gcp, need to investigate.
> * In the inventory, the "storage_cluster_instance_ids": ["6394890111382239406","1037653424555473070","6095985740772587694","674473927763343534"] need to be replaced with self-link.

- ingress rule already exits at line 80, 90 
- removed egress rules for compute
- bypassed 60sec wait by introducing new var wait_for_instance
- changed storage_cluster_instance_ids to self_links

praveenramuu committed:
updated security output variables and  egress enablement based on cluster type

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Rmoved tags variables and added storage internal ingress

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Added data source for fetching cidr range from subnet name

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Added port range 60000-61000 and used resource_prefix as firewall_name_prefix

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Merge branch 'next_gen' into firewall_tag_to_cidr

praveenramuu committed:
Added readMe update and ported ansible_configuration turn on changes

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Moved allow_traffic_bastion_scale_cluster to bastion

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
removed vpc_cluster_public_subnets var

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Added comments to modules

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Adding destination_ranges support , hence re-defining firewall rule

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Merge branch 'IBM:next_gen' into firewall_tag_to_cidr

praveenramuu committed:
Adding using_direct_connection and using_jumphost_connection var

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Bastion changes- added intenals and naming changes , removed allow_traffic_bastion_to_scale_cluster

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Bastion changing rule naming conventions

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Adding auto fetch public cidr data source and rule allow_traffic_bastion_to_scale_cluster

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>

praveenramuu committed:
Add auto identiyfing bastion subnet by datasource for allow_traffic_bastion_to_scale_cluster

Signed-off-by: Praveen Ramu <praveen.ramu@ibm.com>
